src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -293,14 +294,21 @@ uiModules
 
   function transferVisState(fromVis, toVis, stage) {
     return function () {
+
+      const fromVisLib = getVisLibOptions(fromVis);
+      const toVisLib = getVisLibOptions(toVis);
+
       const view = fromVis.getEnabledState();
       const full = fromVis.getState();
       toVis.setState(view);
       editableVis.dirty = false;
       $state.vis = full;
-      $state.save();
 
-      if (stage) $scope.fetch();
+      if (stage && _.isEqual(toVisLib, fromVisLib)) {//only requery ES when the query changed. vislib params are irrelevant
+        $scope.fetch();
+      } else {
+        $state.save();
+      }
```

Comment:
Small suggestion. I think we make this a little more readable by nesting conditions, exiting early, and putting comments on their own line:

```javascript
if (stage) {
  // Only require ES when the query has changed. vislib params are irrelevant.
  if (_.isEqual(toVisLib, fromVisLib) {
    return $scope.fetch();
  }
}

$state.save();
```